src/npc/npcs_state_registry.py

```diff
@@ -0,0 +1,172 @@
+
+    def get_ingroup_deaths_amount(self) -> int:
+        return self._count_dead_npcs_by_study_group(StudyGroup.INGROUP)
+
+    def get_outgroup_deaths_amount(self) -> int:
+        return self._count_dead_npcs_by_study_group(StudyGroup.OUTGROUP)
+
+    def get_total_deaths_amount(self) -> int:
+        return self.get_outgroup_deaths_amount() + self.get_ingroup_deaths_amount()
+
+    def set_current_map_name(self, current_map: str | Map):
+        self.current_map_name = (
+            current_map.value if isinstance(current_map, Map) else current_map
+        )
+
+    def is_npc_dead(self, npc_id: int, study_group: StudyGroup) -> bool:
+        result: bool = False
+        for map_name in self.data.keys():
+            current_map_data: dict[str : list[[int]]] = self.data[map_name]
+            if study_group.name not in current_map_data.keys():
+                continue
+            dead_npcs_list: list[dict] = current_map_data[study_group.name]
+            dying_npc_state = self._get_state(dead_npcs_list, npc_id)
+            if dying_npc_state is not None:
+                result = (
+                    dying_npc_state[NPC_DEAD] if NPC_DEAD in dying_npc_state else False
+                )
+                break
+        return result
+
+    def enable(self) -> None:
+        self.enabled = True
+
+    def disable(self) -> None:
+        self.enabled = False
+
+    def is_enabled(self) -> bool:
+        return self.enabled
+
+    def _is_not_ready_to_update(self):
+        return (
+            self.get_current_time() - self.last_health_update_timestamp
+        ).total_seconds() < HEALTH_UPDATE_DELTA_SECONDS
+
+    def _get_npcs_state_list(self, study_group: StudyGroup) -> list[dict]:
+        map_data: dict[str : list[[dict]]] = (
+            self.data[self.current_map_name]
+            if self.current_map_name in self.data.keys()
+            else None
+        )
+        if map_data is None:
+            map_data = {}
+            self.data[self.current_map_name] = map_data
+        return self._get_npc_states(map_data, study_group)
+
+    def _get_npc_states(
+        self, map_data: dict[str, list[dict]], study_group: StudyGroup
+    ) -> list[dict]:
+        study_group_data: list[dict] = (
+            map_data[study_group.name] if study_group.name in map_data.keys() else None
+        )
+        if study_group_data is None:
+            study_group_data = []
+            map_data[study_group.name] = study_group_data
+
+        return study_group_data
+
+    def _get_state(self, npc_states: list[dict], npc_id: int) -> dict:
+        current_state: dict = self._get_npc_state_data(npc_states, npc_id)
+        if current_state is None:
+            current_state = {NPC_ID: npc_id}
+            npc_states.append(current_state)
+        return current_state
+
+    def _get_npc_state_data(self, npc_states, npc_id: int) -> dict | None:
+        if npc_states is None or len(npc_states) == 0:
+            return None
```

Comment:
This is just a comment, but writing `return` works identically to specifying `None` when returning from a function.
Functions that don't have a return statement always return `None`